app/controllers/projects/invites_controller.rb

```diff
@@ -0,0 +1,30 @@
+# Invite contributors to a Project
+class Projects::InvitesController < Projects::BaseController
+  before_action :authenticate
+
+  def create
+    if @project.contributor?(current_user)
+      flash[:notice] = _('You are already a contributor on this project')
+    else
+      @project.contributors << current_user
+      flash[:notice] = _('Welcome to the project!')
+    end
+
+    redirect_to @project
+  end
+
+  private
+
+  def find_project
+    @project = Project.find_by(id: params[:token])
```

Comment:
This is currently finding by ID because I just wanted to get something running – we'll want some sort of invite token here.